dogstatsd/entrypoint.sh

```diff
@@ -17,11 +17,19 @@ if [[ $DD_URL ]]; then
 fi
 
 # ensure that the trace-agent doesn't run unless instructed to
-export DD_APM_ENABLED=false
 if [[ $DD_APM_ENABLED ]]; then
   export DD_APM_ENABLED=${DD_APM_ENABLED}
 fi
```

Comment:
> ensure that the trace-agent doesn't run unless instructed to

this captures the original intention of the thing even though the implementation was botched 🙈 

since this is the dogstatsd - only image, what i think we want is to _not_ run the trace-agent unless `DD_APM_ENABLED=true`. This is complicated by the fact that the agent itself will [default to running](https://github.com/DataDog/datadog-trace-agent/blob/master/config/agent.go#L156) unless told not to. So I think what we need is something like:

```
 if [[ $DD_APM_ENABLED ]]; then
    export DD_APM_ENABLED=${DD_APM_ENABLED}
 else 
    # disable the agent when the env var is absent
    export DD_APM_ENABLED=false
 fi
```

Does that make sense?